Implemented Spectral Slingshot 实现了幻灵弹弓

CgAu3 opened the pull request:
- Resolved #2379 
注意有不少渲染代码
以及SpectralProjectile是幻灵弹弓射出的弹射物

CgAu3 committed:
Added Spectral Slingshot and Spectral Projectile, but not implemented

CgAu3 committed:
Continued on Spectral Slingshot

CgAu3 committed:
Modified 3rd person arms for spectral slingshot, made Quick Charge work for its cd

CgAu3 committed:
Fixed the ammo draw of spectral slingshot, added properties for its projectile

CgAu3 committed:
Made Spectral Projectile entity render.

CgAu3 committed:
Modified damage and render of spectral projectile (draft)

CgAu3 committed:
Implemented Spectral Projectile damage calculation and Enchantments

CgAu3 committed:
Made the translucency of spectral projectile work

CgAu3 committed:
Fixed typo, implemented spectral slingshot item custom rendering

CgAu3 committed:
Spectral Slingshot can now only load items with melee damage

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/client/renderer/entity/SpectralProjectileRenderer.java:
```diff
@@ -0,0 +1,186 @@
+                poseStack.translate(
+                    FLAT_ITEM_BUNDLE_OFFSET_X * groundScaleX,
+                    FLAT_ITEM_BUNDLE_OFFSET_Y * groundScaleY,
+                    FLAT_ITEM_BUNDLE_OFFSET_Z * groundScaleZ);
```
此处“)”前应换行

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/client/renderer/item/SpectralSlingshotRenderer.java:
```diff
@@ -0,0 +1,140 @@
+                     itemStack,
+                     combinedLight,
+                     combinedOverlay,
+                     poseStack, vertexconsumer)
```
请于 114 行 “(” 之后， 118 行 “,” 之后， 118 行 “)” 之前换行并调整缩进

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/client/renderer/item/WrappingItemCustomRenderedModels.java:
```diff
@@ -0,0 +1,51 @@
+        //Method copied from Create Mod
+        @Override
+        public BakedModel applyTransform(ItemDisplayContext cameraItemDisplayContext, PoseStack mat,
+                                         boolean leftHand) {
```
请于 43行 “(” 之后， 43 行 “,” 之后， 44 行 “)” 之前换行并调整缩进

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/entity/SpectralProjectileEntity.java:
```diff
@@ -0,0 +1,192 @@
+        DamageSource damagesource = this.damageSources().arrow(this, (entity1 != null ? entity1 : this));
+        DamageSource meleeSource =
+            entity1 instanceof Player p1 ? entity1.damageSources().playerAttack(p1) :
+                (entity1 instanceof LivingEntity livingEntity ? entity1.damageSources().mobAttack(livingEntity) : damagesource);
```
请于 106 行 “?” 之前，请于 106 行 “:” 之前， 107 行 “（” 之后，请于 107 行 “?” 之前，请于 107 行 “:” 之前， 107 行 “)” 之前换行并调整缩进

CgAu3 committed:
调整代码格式

Gu-ZT committed:
refactor(entity): 优化 Style

- 调整注释格式并澄清双关语含义
- 移除未使用的随机数生成器
- 更新实体命中逻辑中的伤害源处理
- 重构顶点消费者为记录类以提高可读性
- 优化弹弓物品装填与射击逻辑
- 改进客户端渲染器中的模型变换处理
- 清理冗余代码并统一注释风格
